contracts/exchangeIssuance/DEXAdapter.sol

```diff
@@ -188,6 +182,90 @@ library DEXAdapter {
         }
     }
 
+    /**
+     * Gets the output amount of a token swap.
+     *
+     * @param _swapData     the swap parameters
+     * @param _addresses    Struct containing relevant smart contract addresses.
+     * @param _amountIn     the input amount of the trade
+     *
+     * @return              the output amount of the swap
+     */
+    function getAmountOut(
+        Addresses memory _addresses,
+        SwapData memory _swapData,
+        uint256 _amountIn
+    )
+        external
+        returns (uint256)
+    {
+        if (_swapData.path[0] == _swapData.path[_swapData.path.length-1]) {
+            return _amountIn;
+        }
+
+        if (_swapData.exchange == Exchange.UniV3) {
+            return _getAmountOutUniV3(_swapData, _addresses.uniV3Quoter, _amountIn);
+        } else if (_swapData.exchange == Exchange.Curve) {
+            (int128 i, int128 j) = _getCoinIndices(
+                _swapData.pool,
+                _swapData.path[0],
+                _swapData.path[1],
+                ICurveAddressProvider(_addresses.curveAddressProvider)
+            );
+            return _getAmountOutCurve(_swapData.pool, i, j, _amountIn, _addresses);
+        } else {
+            return _getAmountOutUniV2(
+                _swapData,
+                IUniswapV2Router02(
+                    (_swapData.exchange == Exchange.Quickswap) ? _addresses.quickRouter : _addresses.sushiRouter
```

Comment:
Now that we use this check in so many places we might want to move it into a `pure` helper function, for better maintainability / readability.